kratos/tests/test_geometries.py

```diff
@@ -1,72 +1,234 @@
+        nodes.append(node1)
+        nodes.append(node2)
+        nodes.append(node3)
+        nodes.append(node4)
+
+        knots = KM.Vector(6)
+        knots[0] = 0.0
+        knots[1] = 0.0
+        knots[2] = 0.0
+        knots[3] = 11.180339887498949
+        knots[4] = 11.180339887498949
+        knots[5] = 11.180339887498949
+
+        curve = KM.NurbsCurveGeometry3D(
+            nodes, 3, knots)
+
+        # check nodes
+        self.assertAlmostEqual(curve[0].X, 0.0)
+        self.assertAlmostEqual(curve[1].X, 3.3333333333333335)
+
+        # checking right pointers/ sources of nodes
+        curve[1].X = 2
+        self.assertAlmostEqual(curve[1].X, node2.X)
+        self.assertAlmostEqual(node2.X, 2)
+
+        # check knot spans
+        self.assertAlmostEqual(curve.Knots()[4], 11.180339887498949)
+        self.assertEqual(curve.NumberOfKnots(), 6)
+        self.assertEqual(curve.NumberOfControlPoints(), 4)
+
+        # check rational
+        self.assertFalse(curve.IsRational())
+
+        # check general information
+        self.assertEqual(curve.Dimension(), 1);
+        self.assertEqual(curve.WorkingSpaceDimension(), 3);
+        self.assertEqual(curve.LocalSpaceDimension(), 1);
+
+    def test_nodes_vector_iterators(self):
+        current_model = KM.Model()
+        model_part= current_model.CreateModelPart("Main")
+
+        node1 = model_part.CreateNewNode(1, 0, 0, 0)
+        node2 = model_part.CreateNewNode(2, 3.3333333333333335, 1.6666666666666667, 0)
+        node3 = model_part.CreateNewNode(3, 6.6666666666666661, 3.333333333333333, 0)
+        node4 = model_part.CreateNewNode(4, 10, 5, 0)
+
+        nodes = KM.NodesVector()
+        nodes.append(node1)
+        nodes.append(node2)
+        nodes.append(node3)
+        nodes.append(node4)
+
+        for node in nodes:
+            node.X = node.X + 1
+
+        # check new coordinates
+        self.assertAlmostEqual(nodes[0].X, node1.X)
+        self.assertAlmostEqual(nodes[0].X, 1.0)
+
+        # check length of nodes vector
+        self.assertEqual(len(nodes), 4)
+
+    def test_nodes_vector_geometry_iterators(self):
+        current_model = KM.Model()
+        model_part= current_model.CreateModelPart("Main")
+
+        node1 = model_part.CreateNewNode(1, 0, 0, 0)
+        node2 = model_part.CreateNewNode(2, 3.3333333333333335, 1.6666666666666667, 0)
+        node3 = model_part.CreateNewNode(3, 6.6666666666666661, 3.333333333333333, 0)
+        node4 = model_part.CreateNewNode(4, 10, 5, 0)
+
+        nodes = KM.NodesVector()
+        nodes.append(node1)
+        nodes.append(node2)
+        nodes.append(node3)
+        nodes.append(node4)
+
+        knots = KM.Vector(6)
+        knots[0] = 0.0
+        knots[1] = 0.0
+        knots[2] = 0.0
+        knots[3] = 11.180339887498949
+        knots[4] = 11.180339887498949
+        knots[5] = 11.180339887498949
+
+        curve = KM.NurbsCurveGeometry3D(
+            nodes, 3, knots)
+
+        for node in curve:

```

Comment:
@maceligueta it is used here, curve is a geometry

Reply:
Not exactly what I was saying, but fair enough. I mean, there won't be any example showing that you can loop over elements and then loop over the nodes of an element. It is fine, I will put it myself in the core. Let's not add noise to this one.